client/approval_policy.go

```diff
@@ -0,0 +1,77 @@
+package client
+
+import "fmt"
+
+type ApprovalPolicy struct {
+	Id                   string           `json:"id"`
```

Comment:
not sure if possible with golang, but can this type be derived from the type of a tamplate? I guess the fields are exactly the same so its basically code duplication. not mandatory if this is the way we are usually doing things

Reply:
There are some ways. But generally, there's no inheritance.
Applying that change will add more changes to an already large PR.